src/mapgen_functions.cpp

```diff
@@ -275,6 +276,17 @@ bool mapgendata::is_groundcover( const ter_id iid ) const {
     return false;
 }
 
+bool mapgendata::has_basement() const
+{
+    std::vector<std::string> all_basements = region.city_spec.basements.all;
+    for( auto b : all_basements ) {
```

Comment:
It won't be possible to use `t_below` unless inside of `any_of` unless I put value of `t_below` to square brackets which is only possible for variables, therefore it will look like this to compile successfully:

```c++
bool mapgendata::has_basement() const
{
    std::vector<std::string> all_basements = region.city_spec.basements.all;
    oter_id what_is_below = t_below;
    return std::any_of( all_basements.begin(),
    all_basements.end(), [what_is_below]( const std::string & b ) {
        return what_is_below == oter_id( b );
    } );
}
```

Reply:
I see. Thanks.